test/docker-e2e/e2e_simple_test.go

```diff
@@ -122,3 +126,30 @@ func testBankSend(t *testing.T, chain *tastoradockertypes.Chain, cfg *dockerchai
 	err = wait.ForBlocks(ctx, 2, chain)
 	require.NoError(t, err, "failed to wait for blocks after transaction")
 }
+
+// createOrGetWalletAddress tries to create a wallet with the given name, but if it already exists,
+// it retrieves the existing wallet's address. This makes the function safe to call multiple times.
+func createOrGetWalletAddress(ctx context.Context, chain *tastoradockertypes.Chain, walletName string) (string, error) {
```

Comment:
instead of implementing this, to me it seems easier to just pass in a new wallet name to `testBankSend` that is dynamically generated so it is always unique.